infra/templates/az-isolated-service-single-region/tests/unit/unit_test.go

```diff
@@ -222,10 +157,8 @@ func TestTemplate(t *testing.T) {
 		TfOptions:             tfOptions,
 		Workspace:             workspace,
 		PlanAssertions:        nil,
-		ExpectedResourceCount: 62,
+		ExpectedResourceCount: 63,
 		ExpectedResourceAttributeValues: infratests.ResourceDescription{
-			// "module.keyvault.azurerm_key_vault.keyvault":                                   expectedKeyVault,
```

Comment:
Are we explicitly removing both unit and integration tests that validate the IP whitelists?

Can you provide some context on why? I understand not having both, as it is redundant, but it would be nice to have at least one test that validates it is configured properly.

Reply:
@nmiodice the IP addresses used as a part of the build are rather ephemeral, and I don't think they are relevant to int/unit testing.

The notion that we need to be able to test that setting an IP results in that IP being present in the white-list makes sense. We hit some problems with sorting out the check needed to work, however. 

Given the overall nature of the work, and the benefit to enabling the build agent pool, I'd like to treat the need to re-establish these particular tests as a new, small, targeted bug/issue.